plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -589,6 +634,15 @@ export async function createRouter(
       }
 
       const createdBy = parseStringsParam(req.query.createdBy, 'createdBy');
+      const [userEntityRef] = [createdBy].flat();
+      await checkTaskPermission({
```

Comment:
mmm there is no task at this point 🤔

I'm not sure how this logic works, but ideally we would need to filter out only the tasks that the user shouldn't be able to see

Reply:
Ah so the idea as explained above is that we're checking whether the `createdBy` filter specified by the user is allowed before make the database query. If they're allowed to use the `userEntityRef` specified in the `createdBy` query, then we could just use the built-in `createdBy` filter to restrict the list of tasks the user is able to see.

Reply:
I see! This works right now, but it won't work in the future if we introduce more task rules that needs the task data.
This makes things a bit tricky in case `createdBy` contains more than one item, as highlighted in your TODO.
Here we have two choices, either we transform the rule in a db query, or we authorize the items one by one below.

Otherwise if we want to keep this "optimization", we could conditionally authorize the permission and then in case we receive a CONDITIONAL response, we manually check whether the response contains a `IS_TASK_OWNER` rule and in that case we perform authorization manually. But this makes things more tricky, so I'm not sure this is worth.

Reply:
Ah ok that makes alot of sense, I'll make the changes.